tools/aux/tpm2_print.c

```diff
@@ -0,0 +1,336 @@
+    if (!res) {
+        goto read_error;
+    }
+    print_yaml_indent(indent_count);
+    tpm2_tool_output("resetCount: %lu\n", (long unsigned int)numb.u32);
+
+    res = files_read_32(fd, &numb.u32);
+    if (!res) {
+        goto read_error;
+    }
+    print_yaml_indent(indent_count);
+    tpm2_tool_output("restartCount: %lu\n", (long unsigned int)numb.u32);
+
+    res = files_read_bytes(fd, &numb.u8, 1);
+    if (!res) {
+        goto read_error;
+    }
+    print_yaml_indent(indent_count);
+    tpm2_tool_output("safe: %u\n", (unsigned int)numb.u8);
+
+    // success
+    return true;
+
+read_error:
+    FILE_2_SHORT_ERROR();
+    return false;
+}
+
+static bool print_TPMS_QUOTE_INFO(FILE* fd, size_t indent_count) {
+    bool res;
+
+    // read TPML_PCR_SELECTION count (UINT32)
+    UINT32 pcr_selection_count;
+    res = files_read_32(fd, &pcr_selection_count);
+    if (!res) {
+        goto read_error;
+    }
+    print_yaml_indent(indent_count);
+    tpm2_tool_output("pcrSelect:\n");
+
+    print_yaml_indent(indent_count + 1);
+    tpm2_tool_output("count: %lu\n", (long unsigned int)pcr_selection_count);
+
+    print_yaml_indent(indent_count + 1);
+    tpm2_tool_output("pcrSelections:\n");
+
+    // read TPML_PCR_SELECTION array (of size count)
+    for(long unsigned int i = 0; i < pcr_selection_count; ++i) {
+        print_yaml_indent(indent_count + 2);
+        tpm2_tool_output("%lu:\n", i);
+
+        // print hash type (TPMI_ALG_HASH)
+        UINT16 hash_type;
+        res = files_read_16(fd, &hash_type);
+        if (!res) {
+            goto read_error;
+        }
+        res = tpm2_alg_util_is_hash_alg(hash_type);
+        if (!res) {
+            LOG_ERR("%s: Invalid hash type in quote", file_path);
+            goto error;
+        }
+        const char* const hash_name = tpm2_alg_util_algtostr(hash_type);
+        print_yaml_indent(indent_count + 3);
+        tpm2_tool_output("hash: %u (%s)\n", (unsigned int)hash_type, hash_name);
+
+        UINT8 sizeofSelect;
+        res = files_read_bytes(fd, &sizeofSelect, 1);
+        if (!res) {
+            goto read_error;
+        }
+        print_yaml_indent(indent_count + 3);
+        tpm2_tool_output("sizeofSelect: %u\n", (unsigned int)sizeofSelect);
+
+        // print PCR selection in hex
+        print_yaml_indent(indent_count + 3);
+        tpm2_tool_output("pcrSelect: ");
+        res = tpm2_util_hexdump_file(fd, sizeofSelect, true);
+        tpm2_tool_output("\n");
+        if (!res) {
+            goto read_error;
+        }
+    }
+
+    // print digest in hex (a TPM2B object)
+    print_yaml_indent(indent_count);
+    tpm2_tool_output("pcrDigest: ");
+    res = tpm2_util_print_tpm2b_file(fd);
+    tpm2_tool_output("\n");
+    if (!res) {
+        goto read_error;
+    }
+
+    // success
+    return true;
+
+read_error:
+    FILE_2_SHORT_ERROR();
+error:
+    return false;
+}
+
+static bool print_TPMS_ATTEST_yaml(FILE* fd, size_t indent_count) {
+    bool res;
```

Comment:
declare and assign at first use on 196 per the coding standard.

Reply:
Fixed